configure.ac

```
@@ -10,8 +10,123 @@ AM_INIT_AUTOMAKE([foreign subdir-objects])
+		[OPENSSL_inc_check="$with_openssl_include $with_openssl_include/include $with_openssl_include/include/ssl"])
+
+	dnl
+	dnl Look for OpenSSL library
+	dnl
+	AC_CACHE_CHECK([for OpenSSL library location], [ac_cv_openssl_lib],
+	[
+		for dir in $OPENSSL_lib_check
+		do
+			if test -d "$dir" && \
+				( test -f "$dir/libssl.so" ||
+				  test -f "$dir/libssl.a"  ||
+				  test -f "$dir/${host_cpu}-${host_os}/libssl.so" ||
+				  test -f "$dir/${host_cpu}-${host_os}/libssl.a" )
+			then
+				ac_cv_openssl_lib=$dir
+				break
+			fi
+		done
+
+		if test -z "$ac_cv_openssl_lib"
+		then
+			AC_MSG_ERROR([Didn't find the OpenSSL library dir in '$OPENSSL_lib_check'])
+		fi
+
+		case "$ac_cv_openssl_lib" in
+			/* ) ;;
+			* )  AC_MSG_ERROR([The OpenSSL library directory ($ac_cv_openssl_lib) must be an absolute path.]) ;;
+		esac
+	])
+	AC_SUBST([OPENSSL_LIB_DIR],[$ac_cv_openssl_lib])
+
+	dnl
+	dnl Look for OpenSSL header file directory
+	dnl
+	AC_CACHE_CHECK([for OpenSSL include path], [ac_cv_openssl_inc],
+	[
+		for dir in $OPENSSL_inc_check
+		do
+			if test -d "$dir" && test -f "$dir/openssl/ssl.h"
+			then
+				ac_cv_openssl_inc=$dir
+				break
+			fi
+		done
+
+		if test -z "$ac_cv_openssl_inc"
+		then
+			AC_MSG_ERROR([Didn't find the OpenSSL header dir in '$OPENSSL_inc_check'])
+		fi
+
+		case "$ac_cv_openssl_inc" in
+			/* ) ;;
+			* )  AC_MSG_ERROR([The OpenSSL header directory ($ac_cv_openssl_inc) must be an absolute path.]) ;;
+		esac
+	])
+	AC_SUBST([OPENSSL_INC_DIR],[$ac_cv_openssl_inc])
+
+	dnl
+	dnl Now check that the above checks resulted in -I and -L flags that
+	dnl let us build actual programs against OpenSSL.
+	dnl
+	case "$ac_cv_openssl_lib" in
+	  /usr/lib) ;;
+	  *) LDFLAGS="$LDFLAGS -L${ac_cv_openssl_lib}" ;;
+	esac
+	CPPFLAGS="$CPPFLAGS -I${ac_cv_openssl_inc}"
+	AC_MSG_CHECKING([that we can build OpenSSL programs])
+	AC_COMPILE_IFELSE(
+		[AC_LANG_PROGRAM([#include <openssl/ssl.h>],
+			[init_openssl_library()])],
```

I have noticed that the init function should read SSL_library_init() - and also this check seems not to work. It silently goes passes and just writes out a warning into config.log:
> warning: implicit declaration of function 'init_openssl_library' [-Wimplicit-function-declaration]